src/Migrations/M033MigrateSocialMediaTwitterBlockToEmbedBlock.php

```diff
@@ -0,0 +1,286 @@
+<?php
+
+// phpcs:disable Generic.Files.LineLength.MaxExceeded
+
+namespace P4\MasterTheme\Migrations;
+
+use P4\MasterTheme\MigrationRecord;
+use P4\MasterTheme\MigrationScript;
+use WP_Block_Parser;
+
+/**
+ * Migrate Social Media blocks with Twitter/X urls to Embed blocks.
+ */
+class M033MigrateSocialMediaTwitterBlockToEmbedBlock extends MigrationScript
+{
+    private const SOURCE_INSTAGRAM = 'instagram';
+    private const SOURCE_FACEBOOK = 'facebook';
+    private const SOURCE_TWITTER = 'twitter';
+    private const SOURCE_X = 'x';
+
+    private const SOURCE_PATTERNS = [
+        self::SOURCE_INSTAGRAM => '/^(?:(?:http|https):\/\/)?(?:www.)?(?:instagram.com|instagr.am|instagr.com)\/.+$/i',
+        self::SOURCE_FACEBOOK => '/^(?:(?:http|https)?:\/\/)?(?:www\.)?(mbasic.facebook|m\.facebook|facebook|fb)\.(com|me)\/.+$/i',
+        self::SOURCE_TWITTER => '/^(https?\:\/\/)?(www\.)?(twitter\.com)\/.+$/i',
+        self::SOURCE_X => '/^(https?\:\/\/)?(www\.)?x\.com\/.+$/i',
+    ];
+
+    /**
+     * Perform the actual migration.
+     *
+     * @param MigrationRecord $record Information on the execution, can be used to add logs.
+     * phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter -- interface implementation
+     */
+    protected static function execute(MigrationRecord $record): void
```

Comment:
This function was completely refactored to make it reusable. You can see the new function [here](https://github.com/greenpeace/planet4-master-theme/blob/0855240dd8f336678a6ead6e97c82d7ed6b3153d/src/Migrations/Utils/Functions.php#L21) and an example of its implementation [here](https://github.com/greenpeace/planet4-master-theme/blob/0855240dd8f336678a6ead6e97c82d7ed6b3153d/src/Migrations/M032MigrateSplit2ColumnBlock.php#L21).

Reply:
Yep I saw that, but in this case there's a bunch of extra `if` cases

Reply:
That's why the `Utils\Functions::execute_block_migration` receives the callable function `$block_check_callback` as a parameter. You should create an extra function with the `if` conditions and pass it as a parameter, something like this:

```php 
public static function execute(MigrationRecord $record): void
    {
        $check_is_valid_block = function ($block) {
            return self::check_is_valid_block($block);
        };

        $transform_block = function ($block) {
            return self::transform_block($block);
        };

        Utils\Functions::execute_block_migration(
            Utils\Constants::BLOCK_SOCIAL_MEDIA,
            $check_is_valid_block,
            $transform_block,
        );
    }
``` 
```php
private static function check_is_valid_block(array $block): bool
    {
        // Check if the block is valid.
                    if (!is_array($block)) {
                        return false;
                    }

                    // Check if the block has a 'blockName' key.
                    if (!isset($block['blockName'])) {
                        return false;
                    }

                    // Check if the block is a Social Media block.
                    if ($block['blockName'] !== Utils\Constants::BLOCK_SOCIAL_MEDIA) {
                        return false;
                    }

                    // Check if the embed is a Facebook page, in which case we do nothing.
                    if (isset($block['attrs']['embed_type']) && $block['attrs']['embed_type'] === 'facebook_page') {
                        return false;
                    }

                    // If not, we get the social media URL.
                    $social_media_url = $block['attrs']['social_media_url'] ?? null;

                    // If there's no social media url, we do nothing.
                    if (!$social_media_url) {
                        return false;
                    }

                    // Identify the source of the media (Facebook, Instagram, Twitter, X)
                    $source = self::identify_source($social_media_url);

                    // If there's no source, or if it's Facebook or Instagram, we do nothing.
                    // We only want to migrate Twitter/X embeds.
                    if (!$source || $source === self::SOURCE_FACEBOOK || $source === self::SOURCE_INSTAGRAM) {
                        return false;
                    }
    }
``

Let me know if it's not clear enough.

Reply:
It was clear, thank you! Updated 👍